sdk/openai/azure-ai-openai-assistants/src/main/java/com/azure/ai/openai/assistants/implementation/Util.java

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.openai.assistants.implementation;
+
+import com.azure.ai.openai.assistants.AssistantsServiceVersion;
+import com.azure.core.http.rest.RequestOptions;
+
+/**
+ * Utility class to be used by the SDK internally.
+ */
+public class Util {
+    /**
+     * This is the endpoint that non-azure OpenAI supports. Currently, it has only v1 version.
+     */
+    private static final String OPEN_AI_ENDPOINT = "https://api.openai.com/v1";
+
+    /**
+     * Get the endpoint for OpenAI service.
+     */
+    public static String getOpenAIEndpoint() {
+        return OPEN_AI_ENDPOINT;
+    }
+
+    /**
+     * OpenAI service can be used by either not setting the endpoint or by setting the endpoint to start with
+     * "https://api.openai.com/v1"
+     */
+    public static boolean useAzureOpenAIService(String endpoint) {
```

Comment:
What happens if the user sets the endpoint to `builder.endpoint("https://api.openai.com")` which is valid as they don't need to set the version to use?

Reply:
if user gives the valid public openai endpoint, this method will return false, which means use non azure open ai client, then this [line](https://github.com/Azure/azure-sdk-for-java/pull/38669/files#diff-5d92a1c5b5fa0c2352299a33825cc485a6d85f01bb4e439bbafb9102135d5726R351) will use the hardcoded OPEN_AI_ENDPOINT = "https://api.openai.com/v1" the endpoint,

Reply:
The user may not set `/v1` when they are setting the endpoint but we are checking if `v1` is part of the endpoint which results in this method returning `false` when it has to be `true`.

Reply:
Ah I got it now. I can add one more conditional check to make sure it returns true when endpoint is `https://api.openai.com/`